docker/Dockerfile.optimism

```diff
@@ -0,0 +1,57 @@
+FROM alpine:3.20 AS foundry-build
+
+ARG TARGETARCH
+WORKDIR /opt
+
+RUN apk add clang lld curl build-base linux-headers git \
+    && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh \
+    && chmod +x ./rustup.sh \
+    && ./rustup.sh -y
+
+RUN [[ "$TARGETARCH" = "arm64" ]] && echo "export CFLAGS=-mno-outline-atomics" >> $HOME/.profile || true
+
+WORKDIR /opt/foundry
+RUN git clone https://github.com/foundry-rs/foundry.git . && git checkout 1bcd17c
+
+# see <https://github.com/foundry-rs/foundry/issues/7925>
+RUN git update-index --force-write-index
+
+RUN --mount=type=cache,target=/root/.cargo/registry --mount=type=cache,target=/root/.cargo/git --mount=type=cache,target=/opt/foundry/target \
+    source $HOME/.profile && cargo build --release --features cast/aws-kms,forge/aws-kms \
```

Comment:
Instead of building the Foundry, can we just run the suggested command on their [website](https://getfoundry.sh)? Otherwise, it takes so long on Docker to build it.

Reply:
Does not work with musl libc. The build setup is copied from their [dockerfile](https://github.com/foundry-rs/foundry/blob/master/Dockerfile). It does not seem that they build & push it anywhere public though, so for now we keep it here.